main/bamboo/bamboo_test.go

```diff
@@ -0,0 +1,207 @@
+package main
+
+import (
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"reflect"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/QubitProducts/bamboo/configuration"
+)
+
+func eventSourceHandler(lines ...string) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusOK)
+
+		for _, l := range lines {
+			fmt.Fprintln(w, l)
+		}
+	})
+}
+
+func TestConnectToMarathonEventStream(t *testing.T) {
+	for _, test := range []struct {
+		desc     string
+		user     string
+		password string
+		handler  http.Handler
+		count    int
+		payloads [][]byte
+	}{
+		{
+			desc: "not-found",
+			handler: http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				w.WriteHeader(http.StatusNotFound)
+			}),
+		},
+		{
+			desc:    "single-payload",
+			handler: eventSourceHandler("data: payload"),
+			count:   1,
+			payloads: [][]byte{
+				[]byte("payload\n"),
+			},
+		},
+		{
+			desc:    "heartbeat",
+			handler: eventSourceHandler("", "", ""),
+			count:   0,
+		},
+		{
+			desc:    "event line",
+			handler: eventSourceHandler("event: eventName"),
+			count:   0,
+		},
+		{
+			desc:    "mixed content",
+			handler: eventSourceHandler("", "event: eventName", "data: payload", "", "", "event: eventName2", "data: payload2", ""),
+			count:   2,
+			payloads: [][]byte{
+				[]byte("payload\n"),
+				[]byte("payload2\n"),
+			},
+		},
+	} {
+		t.Run(test.desc, func(t *testing.T) {
```

Comment:
What go version are we using for Bamboo? Weren't go subtests introduced in 1.8? Or am I being stupid..